src/main/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAO.java

```diff
@@ -67,15 +67,17 @@ public void deleteDataset(Dataset dataset, Integer userId) throws Exception {
   public void deleteStudy(Study study, User user) throws Exception {
     jdbi.useHandle(handle -> {
       handle.getConnection().setAutoCommit(false);
-      study.getDatasets().forEach(d -> {
-        try {
-          deleteDataset(d, user.getUserId());
-        } catch (Exception e) {
-          handle.rollback();
-          logException(e);
-          throw new DatasetDeletionException(e);
-        }
-      });
+      if (study.getDatasets() != null) {
```

Comment:
Could study.getDatasets() alternatively return an empty list if there are no datasets? Not sure if that would be better or not.

Reply:
It's theoretically possible, but the next `forEach` safely skips empty sets.

Reply:
Looks like @snf2ye 's suggestion could likely simplify some other checks where we're both null and empty list checking as well.
If time is of a concern, I'd be comfortable with this change as proposed and adding a ticket to consider the refactoring of this per the suggestion.

Reply:
Ah ... I think I misunderstood the suggestion. Yes, we can modify the getter, that also works.